app/controllers/admin/routes.py

```diff
@@ -91,9 +91,10 @@ def createEvent(templateid, programid=None):
         eventData['contactName'] = "CELTS Admin"
         eventData['contactEmail'] = app.config['celts_admin_contact']
         if program:
-            if program.contactName and program.contactEmail:
+            if program.contactName or program.contactEmail or program.programLocation:
```

Comment:
The issue with doing "or" like this is that if they have a location, but no contactEmail or contactName, it will default to an empty string, this is a problem because just above that line we want to make sure if there is no contactEmail or contactName it defaults to "CELTS Admin" and the default CELTS contact.

Reply:
> 

Made another if statement that detects if program.programLocation is changed. This way contactEmail and contactName will not be empty strings when only program.programLocation is changed.